src/sink.rs

```diff
@@ -105,12 +105,19 @@ impl<T> BoundedSink<T> {
 
 impl<T> Sink<T> for BoundedSink<T> {
     type Error = ();
-    fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        self.sender
-            .poll_ready(cx)
-            .map_err(|error| error!(message = "Sender error.", %error))
+    fn poll_ready(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        use mpsc::error::TrySendError;
+
+        match self.sender.try_reserve() {
+            Ok(_) => Poll::Ready(Ok(())),
+            Err(TrySendError::Full(_)) => Poll::Pending,
```

Comment:
Found the source of the hangups. This returns `Poll::Pending` without waking.

Reply:
I'm not sure how widely used this is, but one option would be to move back towards `futures::channel::mpsc` which implements `Sink` and removes the need for this whole thing.

Reply:
Do you have some context into why `UnboundedSink`/`BoundedSink` were created in the first place over just the regular `UnboundedSender`/`Sender`?

Reply:
Hm, it doesn't seem to implement `Sink` anymore: https://github.com/tokio-rs/tokio/issues/2389.

Reply:
Yes, I _believe_ the purpose was just to give them a `Sink` interface since the tokio channels do not provide that out of the box.

Reply:
Refactored in d5174f3.

Test seem to fail, but at least they run now 😸